### ISSUE TITLE
Proposal: Add "allowChained" option to "no-nested-ternary"

### ISSUE THREAD

**Issue by lamflam** [eslint/eslint#10044]
**What rule do you want to change?**
This change would add an option to the `no-nested-ternary` rule.

**Does this change cause the rule to produce more or fewer warnings?**
By default the change would have no effect. If users opt in to this option, it would provide fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**
This would be a new option called "allowChained" that would have a default value of "false" in order to preserve existing behavior unless users opt in to use this option. 

**Please provide some example code that this change will affect:**
Based on https://medium.com/javascript-scene/nested-ternaries-are-great-361bddd0f340, there's an argument that there is a difference between "nested" and "chained" ternary expressions. This option would allow for chained expressions while still limiting nested expressions.

With `"no-nested-ternary": ["error", { "allowChained": true }]`, this would now be allowed:
```js
// Chained, this would be allowed 
cond1
    ? value1
    : cond2
    ? value2
    : value3
```

This would still be an error:
```js
// Nested, this would not be allowed
cond1
    ? cond2
    ? value2
    : value1
    : value3
```

**What does the rule currently do for this code?**
Currently, chained ternary expressions are not allowed at all by this rule.

**What will the rule do after it's changed?**
If a user sets this option, the rule will no longer warn about chained ternary expressions but will still warn for nested ternary expressions.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal. Personally, I'm not sure this would be worthwhile because I think most users who enable `no-nested-ternary` would want to disallow "chaining" too. But I'm not opposed to adding it if there are other team members that think it would be worthwhile.

**Comment by lamflam** [comment]
Thanks for responding @not-an-aardvark. I'm also unsure of exactly how many users would enable the option. I know my team and a few other teams that have discussed using an option like this and thought maybe it go into core instead of our custom eslint plugin since the it doesn't affect current behavior and the implementation is trivial.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.

**Comment by lamflam** [comment]
Unfortunate but understood, thank you for the reply @not-an-aardvark. Leaving a link to [own implementation](https://github.com/lolatravel/eslint-plugin-lola/blob/master/lib/rules/no-nested-ternary.js) of this rule for any other users looking for this functionality.
